yt_dlp/extractor/chzzk.py

```diff
@@ -146,19 +146,39 @@ def _real_extract(self, url):
         video_meta = self._download_json(
             f'https://api.chzzk.naver.com/service/v3/videos/{video_id}', video_id,
             note='Downloading video info', errnote='Unable to download video info')['content']
-        formats, subtitles = self._extract_mpd_formats_and_subtitles(
-            f'https://apis.naver.com/neonplayer/vodplay/v1/playback/{video_meta["videoId"]}', video_id,
-            query={
-                'key': video_meta['inKey'],
-                'env': 'real',
-                'lc': 'en_US',
-                'cpl': 'en_US',
-            }, note='Downloading video playback', errnote='Unable to download video playback')
+        video_status = video_meta.get('vodStatus')
+        video_live_date = video_meta.get('liveOpenDate')
+        if video_live_date:
+            live_status = 'was_live'
+        else:
+            live_status = 'not_live'
 
+        if video_status == 'UPLOAD':
+            playback = self._parse_json(video_meta.get('liveRewindPlaybackJson'), video_id)
+            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+                traverse_obj(playback, ('media', 0, 'path')), video_id,
+                note='Downloading video playback', errnote='Unable to download video playback')
```

Comment:
since the `_extract_m3u8_formats` call is ultimately fatal, we don't need to use `.get()`/`traverse_obj` on the way there:
```suggestion
            playback = self._parse_json(video_meta['liveRewindPlaybackJson'], video_id)
            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
                playback['media'][0]['path']), video_id, 'mp4', m3u8_id='hls')
```